.github/workflows/CI.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  Test:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+          - windows-2019
+          - macos-10.15
+        d:
+          - "ldc-1.27.1"
+          - "dmd-2.097.2"
+        meson:
+          - 0.59.1
+        ninja:
+          - 1.10.2
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      # Cache
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.dub
+            ~/AppData/Local/dub
```

Comment:
sorry, was refering to all the dub folders with the dub cache comment

Reply:
Do you have a well-defined reproduction of the bug you are talking about? I haven't faced that so far. If that's the case, this is a serious issue that Dub needs to fix ASAP.

Reply:
when not checking in the dub.selections.json I think it can happen that it will reuse outdated dependencies from the cache if you don't run `dub upgrade`, though the issues I faced I usually didn't investigate for the actual cause, but went with `dub upgrade` to get the dependencies I actually want and specified in the dub recipe instead.

Reply:
There are no dub.selections.json files in this repo. So, I don't think this bug is relevant here.

Reply:
yes that's exactly when issues with updating dependency versions in dub.sdl/json while having local cache has arisen in the past for me